swayfmt/src/items/item_struct/mod.rs

```diff
@@ -69,9 +69,17 @@ impl Format for ItemStruct {
                             .map(|(type_field, comma_token)| (&type_field.value, comma_token))
                             .collect::<Vec<_>>();
                         // In first iteration we are going to be collecting the lengths of the struct variants.
+                        // We need to include the `pub` keyword in the length, if the field is public,
+                        // together with one space character between the `pub` and the name.
                         let variant_length: Vec<usize> = value_pairs
                             .iter()
-                            .map(|(type_field, _)| type_field.name.as_str().len())
+                            .map(|(type_field, _)| {
+                                type_field
+                                    .visibility
+                                    .as_ref()
+                                    .map_or(0, |_| PubToken::AS_STR.len() + 1)
```

Comment:
Just thinking if we ever had other types of visibility tokens, we shouldn't hardcode this.

```suggestion
                                    .map_or(0, |token| token.span().as_str().len() + 1)
```